test/functional/subnet/create_test.rb

```diff
@@ -0,0 +1,184 @@
+require_relative '../test_helper'
+require 'hammer_cli_foreman/subnet'
+
+module HammerCLIForeman
+  describe Subnet do
+    describe CreateCommand do
+      it 'allows minimal options' do
+        api_expects(:subnets, :create) do |par|
+          par['subnet']['name'] == 'net1'
+        end
+        run_cmd(%w(subnet create --name net1 --network=192.168.122.0 --mask=255.255.255.0))
+      end
+
+      it 'allows dhcp id' do
+        api_expects(:subnets, :create).with_params({
+          :subnet => {
+            :name => 'net1',
+            :dhcp_id => '1'
+          }
+        })
+        run_cmd(%w(subnet create --name net1 --network=192.168.122.0 --mask=255.255.255.0 --dhcp-id 1))
+      end
+
+      it 'allows dhcp name' do
+        api_expects(:smart_proxies, :index) do |p|
+          p[:search] = "name = \"sp1\""
+        end.returns(index_response([{'id' => 1}]))
+        api_expects(:subnets, :create) do |p|
+          p['subnet']['dhcp_id'] == 1 &&
+            p['subnet']['name'] == 'net1'
+        end
```

Comment:
Not that important but just for you to know, you can use `with_params` here:
```ruby
api_expects(:subnets, :create).with_params({
  :subnet => {
    :name => 'net1',
    :dhcp_id => 1
  }
})
```
It performs the same test but the benefit is much more readable error message if it fails.
Compare
```
expected params to match block at: /home/vagrant/hammer/hammer-cli-foreman/test/functional/subnet/create_test.rb:31
```
with output produced by `with_params`
```
  expected params to include: {
    "subnet": {
      "name": "net12",
      "dhcp_id": 1
    }
  }
```

Reply:
One more idea I got (feel free to ignore it if you don't like it) is defining a `subnet_params` helper somewhere up in the test:
```ruby
def subnet_params(additional_params = {})
  params = {
    :subnet => {
      :name => 'net1'
    }
  }
  params[:subnet].merge!(additional_params)
  params
end
```
That would reduce code repetition and enable writing shorter tests like:
```ruby
api_expects(:subnets, :create).with_params(subnet_params(:tftp_id => '1'))
```

But your solution works and I'm fine with so I'm leaving it up to you to consider whether it makes sense to add the helper.

Reply:
Thanks for suggesting the helper method. I would definitely like to consider adding the helper and accordingly writing the tests.